app/logic/spreadsheet.py

```diff
@@ -82,10 +80,10 @@ def getRetentionRate():
     return  retentionDict
 
 def termParticipation(termDescription):
-    participationQuery = (ProgramEvent.select(ProgramEvent.program_id, EventParticipant.user_id.alias('participant'), Program.programName.alias("progName"))
-                                      .join(EventParticipant, JOIN.LEFT_OUTER, on=(ProgramEvent.event == EventParticipant.event))
-                                      .join(Program, on=(Program.id == ProgramEvent.program_id))
-                                      .join(Event, on=(ProgramEvent.event_id == Event.id))
+    participationQuery = (Event.select(Event.program, EventParticipant.user_id.alias('participant'), Program.programName.alias("progName"))
+                                      .join(EventParticipant, JOIN.LEFT_OUTER, on=(Event.id == EventParticipant.event))
+                                      .join(Program, on=(Program.id == Event.program))
+                                      .join(Event, on=(Event.event_id == Event.id))
```

Comment:
You are already selecting from Event, so joining Event in again doesn't add anything.